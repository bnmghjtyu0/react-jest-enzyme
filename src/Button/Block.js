import React, { useState, useEffect } from 'react'
import { withRouter } from 'react-router-dom'
const Block = ({ name, history, location }) => {
    const [a, setA] = useState(name)
    useEffect(() => {
        const params = new URLSearchParams(location.search)
        if (params.has('day')) {
            setA('77777')
        }
    })
    return (
        <div>
            {a}
        </div>

    )
}
export default withRouter(Block)