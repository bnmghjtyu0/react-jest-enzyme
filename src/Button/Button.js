import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Button extends Component {
    constructor(props) {
        super(props)
        this.state = {
            kk: 0,
            cs: -1
        }

    }
    cc() {
        var ul = []
        var li = []
        if (this.state.cs > 0) {
            li.push(<li data-test="component-li" key={0}>123</li>)
            ul.push(<ul key={0}>{li}</ul>)
        } else {
            li.push(<li data-test="component-li" key={0}>444</li>)
            ul.push(<ul key={0}>{li}</ul>)
        }
        return ul
    }

    hh = () => {
        if (this.state.kk > 0) {
            console.log('kk')
        } else {
            console.log('hh')
        }
    }
    mm = (e) => {
        e.preventDefault()
    }

    render() {
        return (
            <div>
                <button data-test="component-button" onClick={this.props.onClose}>取消</button>
                <button data-test="hh-button" onClick={this.hh}>hh</button>
                <button data-test="mm-button" onClick={e => this.mm(e)}>hh</button>
                {this.cc()}
            </div>
        )
    }
}

export default Button