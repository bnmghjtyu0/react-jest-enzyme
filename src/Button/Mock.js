import React, { Component } from 'react'
import PropTypes from 'prop-types'
import axios from 'axios'
class Mock extends Component {
    constructor(props) {
        super(props)
        this.state = {
            counter: 0
        }

    }

    getAllGoods = () => {
        // 使用 axios 中的 get Function 獲得資料
        return axios.get('url/allGoods').then((resp) => {
            return resp.data
        })
    }


    render() {
        // console.log(this.state.counter)
        return (
            <div>
            </div>
        )
    }
}

export default Mock