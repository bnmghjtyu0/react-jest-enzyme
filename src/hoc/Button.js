import React, { Component } from 'react'
import PropTypes from 'prop-types'
import WithLayout from '../containers/WithLayout'

class Button extends Component {
    constructor(props) {
        super(props)
        this.state = {
            counter: 0,
             foo: 'Foo',
        }
    }

    increment = () => {
        let { counter } = this.state
        counter += 1
        this.setState({
            counter
        })
    }
    render() {

        return (
            <div>
                <button data-test="increment-button" onClick={this.increment}>Increment</button>
            </div>
        )
    }
}

export default WithLayout(Button)