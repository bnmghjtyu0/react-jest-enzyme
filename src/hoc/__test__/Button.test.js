import React from 'react'
import { shallow, mount } from "enzyme";
import { findByTestAttr, checkProps, withLayoutStore } from '../../../test/testUtils';
import sinon from 'sinon';
import Button from '../Button'


// 測試 hoc 方法一 : mount + shallow
// test("renders correctly", () => {
//     const wrapper = shallow(<Button />)
//     // console.log(wrapper.debug())
//     const a = mount(wrapper.get(0))
//     console.log(a)
// });

// 測試 hoc 方法二 : 需要搭配 testUtiles
const setup = () => {
    return shallow(<Button />)
}
setup()

describe('', () => {
    test('hoc', () => {
        const wrapper = mount(setup().get(0))
        // console.log(wrapper.debug())
        const increment = sinon.spy();
        const button = findByTestAttr(wrapper, 'increment-button')
        button.simulate('click')
        setTimeout(() => {
            expect(increment.calledOnce).toBe(true)
        }, 1000)
    })
})