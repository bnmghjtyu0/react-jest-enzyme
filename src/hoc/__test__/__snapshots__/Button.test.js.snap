// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "withLayout",
        <div
          className="container-fluid page-body-wrapper"
        >
          <React.Fragment>
            <Button />
          </React.Fragment>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "withLayout",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <React.Fragment>
            <Button />
          </React.Fragment>,
          "className": "container-fluid page-body-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Symbol(react.fragment),
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "withLayout",
          <div
            className="container-fluid page-body-wrapper"
          >
            <React.Fragment>
              <Button />
            </React.Fragment>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "withLayout",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <React.Fragment>
              <Button />
            </React.Fragment>,
            "className": "container-fluid page-body-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
